### ISSUE TITLE
Request for monthly and yearly response time graph

### ISSUE THREAD

**Issue by coolver83** [louislam/uptime-kuma#2477]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

I would like to request to enable monthly and yearly response time graph option.  Currently only 3h,6h,24h and 1 week response time graph enabled.

![uptime kuma chart](https://user-images.githubusercontent.com/94519115/209601204-9e2713de-7d1e-45b0-9509-3e474b413b67.jpg)

### ✔️ Solution

Please add 1 month, 1 year and custom date to view response time graph would be great!

**Comment by philipdistefano** [comment]
Seconding this request, especially past year.

**Comment by ArtyGreen** [comment]
Up! Please make it! We are really need it!

**Comment by fabry00** [comment]
Can we add also the 1 year statistic above the chart as well?
![image](https://github.com/louislam/uptime-kuma/assets/11576418/0b3220a0-663d-415d-9f17-2c55fd2bf321)

**Comment by blaesus** [comment]
I also need this. (As my service is more used in weekends, and I like to compare weekends).
May I ask what's the status of this feature? Is a patch welcome?

**Comment by CommanderStorm** [comment]
> Is a patch welcome

Yes, it would.
Note that currently we are retooling the monitoring system for the upcoming v2.0 release.
See https://github.com/louislam/uptime-kuma/pull/4171 for further details what is currently not yet done.

Our contribution guide is available here: https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md

**Comment by scarolan** [comment]
+1 for this request.  My ISP kind of sucks and I need the historical data to prove how often and how long they go down.

**Comment by gudge25** [comment]
if I set the monitor every 90m sec
my time range is only about  45 min - hope it can be at least one week

<img width="591" alt="Image" src="https://github.com/user-attachments/assets/b29a4213-2801-43da-8966-6c8173404121" />

**Comment by tobias-carlbom** [comment]
Any updates on this?

**Comment by CommanderStorm** [comment]
@gudge25 what you want is a different feature to what is asked in this issue.

This issue is specifically about the dashboard.
The status page is tracked here instead:
- https://github.com/louislam/uptime-kuma/issues/1888
- (and the PR to fix this) https://github.com/louislam/uptime-kuma/pull/5916

> [!TIP]
> If you want to test said PR, you can run
> ```bash 
> npx kuma-pr peaktwilight:master
> ```

**Comment by CommanderStorm** [comment]
@tobias-carlbom yes, the retooling is done. Feel free to send a PR.

Likely this is can build on `getAggregatedBuckets`, but for this issue, the rebucketing-change likely needs to be implemented properly
- https://github.com/louislam/uptime-kuma/pull/5916#discussion_r2157644309

**Comment by RodzynDev** [comment]
Hi, do you have any updates? I'm really looking forward to those features!

**Comment by CommanderStorm** [comment]
Please don't hesitate to test above PR as I noted above.
I don't think we currently have the bug-free-ness we need to merge this as-is, but this may be wrong.

**Comment by tobias-carlbom** [comment]
Is it possible to test this PR with docker? eg by changing docker tag?

**Comment by CommanderStorm** [comment]
Changing docker tags is not the kind of testing we support. In the end that would mean that people would test in prod.

We have the following docker based workflow though:
https://github.com/louislam/uptime-kuma/wiki/Test-Pull-Requests

**Comment by dabesa** [comment]
@CommanderStorm is this PR still waiting for user testing?
Thanks

**Comment by CommanderStorm** [comment]
It got a good amount of testing, seems like in the current state that would cause trouble, so not mergable.

If you want to help fixing the bugs of that implementation however.. ^^

**Comment by MrNickIE** [comment]
Just checking in on this.  I see that it's not merged because of performance issues.  Fair, but is this being worked on as we need at least a 6-month view for data :(

**Comment by CommanderStorm** [comment]
No, the related PR for the status page has correctness issues in some parts as far as I remember.

If you want to look into that PR and summarising what issues are resolved and which remain that would be how to unblock that one.

- https://github.com/louislam/uptime-kuma/pull/5916

If you want to work on integrating the chart, that should not be that much work but testing that performance is OK would be part of that. I don't think someone has tested what is nessesary in this area.

**Comment by peaktwilight** [comment]
@MrNickIE just updated the PR in question, if you want to take a look for correctness feel free!

**Comment by totti4ever** [comment]
This is still undone and open??

**Comment by tobias-carlbom** [comment]
Apparently there is not enough notification services in Uptime Kuma 😂 So this PR have to wait 🙈

**Comment by JannemanDev** [comment]
I also need this one
